### PR TITLE
Ddpb 1014 agree terms for ROLE_PA

### PR DESCRIPTION
**Pull request by elvisciotti**

### CODE REVIEW

File: src/AppBundle/Controller/PaController.php [hunk 1]
```diff
@@ -18,6 +18,11 @@ class PaController extends AbstractController
      */
     public function indexAction(Request $request)
     {
+        $user = $this->getUser();
+        if ($user->getRoleName() == EntityDir\User::ROLE_PA && !$user->isAgreeTermsUse()) {
```
Comment (slizzio) [inline src/AppBundle/Controller/PaController.php]: === considered best practice

File: src/AppBundle/Controller/PaController.php [hunk 2]
```diff
@@ -18,6 +18,11 @@ class PaController extends AbstractController
...
+            throw new \RuntimeException('The PA user didn not agree with terms and conditions');
```
Comment (slizzio) [inline src/AppBundle/Controller/PaController.php]: small typo - did

File: src/AppBundle/Controller/UserController.php [hunk 3]
```diff
@@ -64,6 +63,12 @@ public function activateUserAction(Request $request, $action, $token)
             $template = 'AppBundle:User:passwordReset.html.twig';
         }
 
+        // PA must agree to terms before activating the account
+        // this check happens before activating the account, therefore no need to set an ACL on all the actions
+        if ('activate' == $action && $user->getRoleName()==EntityDir\User::ROLE_PA && !$user->isAgreeTermsUse()) {
```
Comment (slizzio) [inline src/AppBundle/Controller/UserController.php]: ===

File: src/AppBundle/Controller/UserController.php [hunk 4]
```diff
@@ -86,24 +91,15 @@ public function activateUserAction(Request $request, $action, $token)
             $session = $this->get('session');
             $session->set('_security_secured_area', serialize($clientToken));
 
-            //$request = $this->get("request");
-            //$event = new InteractiveLoginEvent($request, $clientToken);
-            //$this->get("event_dispatcher")->dispatch("security.interactive_login", $event);
-
-            // after password reset
-            if ($action == 'password-reset' /*|| $this->get('security.context')->isGranted('ROLE_ADMIN') || $this->get('security.context')->isGranted('ROLE_AD')*/) {
+            if ($action == 'password-reset') {
```
Comment (slizzio) [inline src/AppBundle/Controller/UserController.php]: ===

File: src/AppBundle/Controller/UserController.php [hunk 5]
```diff
@@ -158,6 +153,9 @@ public function activateLinkSentAction(Request $request, $token)
     public function detailsAction(Request $request)
     {
         $user = $this->getUserWithData(['user']);
+        if ($user->getRoleName() == EntityDir\User::ROLE_PA && !$user->isAgreeTermsUse()) {
```
Comment (slizzio) [inline src/AppBundle/Controller/UserController.php]: ===

File: src/AppBundle/Entity/User.php [hunk 6]
```diff
@@ -750,6 +762,25 @@ public function setPaTeamName($paTeamName)
     }
 
     /**
+     * @return bool
+     */
+    public function isAgreeTermsUse()
```
Comment (slizzio) [inline src/AppBundle/Entity/User.php]: hmmm perhaps should be 'hasAgreedTermsOfUse'?

Comment (elvisciotti) [inline src/AppBundle/Entity/User.php]: I'll change it. PHPStorm auto-creation

File: src/AppBundle/Controller/PaController.php (hunk 2 above)
Comment (elvisciotti) [inline src/AppBundle/Controller/PaController.php]: I'll remove the whole check. I've moved it before activation, so the user won't be even able to log and reach that page

**Pull request merged**